programs/clearing_house/src/math/bank_balance.rs

```diff
@@ -210,3 +213,44 @@ pub fn get_balance_value(
     let (value, _) = get_balance_value_and_token_amount(bank_balance, bank, oracle_price_data)?;
     Ok(value)
 }
+
+pub fn check_withdraw_limits(bank: &Bank) -> ClearingHouseResult<bool> {
+    let deposit_token_amount =
+        get_token_amount(bank.deposit_balance, bank, &BankBalanceType::Deposit)?;
+    let borrow_token_amount =
+        get_token_amount(bank.borrow_balance, bank, &BankBalanceType::Borrow)?;
+
+    let max_borrow_token = bank.withdraw_guard_threshold.max(
+        (deposit_token_amount / 6)
+            .max(
+                bank.borrow_token_twap
+                    .checked_add(bank.borrow_token_twap / 5)
+                    .ok_or_else(math_error!())?,
+            )
+            .min(
+                deposit_token_amount
+                    .checked_sub(deposit_token_amount / 10)
+                    .ok_or_else(math_error!())?,
+            ),
+    ); // between ~15-90% utilization with friction on twap
+
+    let min_deposit_token = bank
+        .deposit_token_twap
+        .checked_sub(
+            (bank.deposit_token_twap / 5)
+                .max(bank.withdraw_guard_threshold.min(bank.deposit_token_twap)),
+        )
+        .ok_or_else(math_error!())?;
+    // friction to decrease utilization (if above withdraw guard threshold)
+
+    msg!(
```

Comment:
do you always wanna log these or just in the failure case? maybe we just log em when `deposit_token_amount >= min_deposit_token && borrow_token_amount <= max_borrow_token` is false